### ISSUE TITLE
Some differences as paper stated

### ISSUE THREAD

**Issue by leftthomas** [euwern/proxynca_pp#2]
1.Why proxies should be divided by 8 at first:
https://github.com/euwern/proxynca_pp/blob/a8e2aa93024f1202ea3253b992c7031696d980d3/loss.py#L24
2.The `lr` is not same as paper stated
https://github.com/euwern/proxynca_pp/blob/a8e2aa93024f1202ea3253b992c7031696d980d3/config/sop.json#L43
https://github.com/euwern/proxynca_pp/blob/a8e2aa93024f1202ea3253b992c7031696d980d3/config/sop.json#L46
https://github.com/euwern/proxynca_pp/blob/a8e2aa93024f1202ea3253b992c7031696d980d3/config/inshop.json#L43
https://github.com/euwern/proxynca_pp/blob/a8e2aa93024f1202ea3253b992c7031696d980d3/config/inshop.json#L46
3.Why use `eps=1.0` in `Adam`:
https://github.com/euwern/proxynca_pp/blob/a8e2aa93024f1202ea3253b992c7031696d980d3/config/inshop.json#L47
4.Why there are `clip grad` step in the training loop:
https://github.com/euwern/proxynca_pp/blob/a8e2aa93024f1202ea3253b992c7031696d980d3/train.py#L416

**Comment by euwern** [comment]
@leftthomas  Thank you for pointing out differences in the paper w.r.t learning rates. It is a typo error and it has been updated in the paper. With regards to the remaining magic numbers, (a) dividing proxies by 8, (b) gradient clipping, (c) eps of adam is set to 1.0. These are the coding artifact that remains behind after many iterations over many months. Based on my investigation, removing (a) and (b) will not affect performance but the model is very sensitive to (c).

**Comment by leftthomas** [comment]
@euwern, the model is very sensitive to (c) is not make sense, using a large `eps` is unreasonable.

**Comment by leftthomas** [comment]
@euwern why close this issue? Can you explain the `eps` setting? This is not reasonable, this makes your paper unreliable

**Comment by euwern** [comment]
This issue is closed, as eps is a hyperparameter. Changing eps will affect the learning rate and other moment vectors in adam optimizer, therefore it will affect the results. With regards to reliability, in what aspect are you measuring this? How would you define the reliability of a given hyperparameter? (e.g.: What is a reasonable learning rate? Why do you think it is reasonable? What makes it reliable?)
Perhaps you could do a study and share your findings.

**Comment by leftthomas** [comment]
@euwern use `eps` as hyperparameter is not common in research, `eps` in `Adam` is used to avoid division by zero, not used to `affect the learning rate`, and I have tested, `eps` is the biggest influence factor with the results, around 20%, however, you didn't mention this point at all in the paper. So how can you proof the results are obtained  because your method works or just because the fine-tuned `eps`?

**Comment by euwern** [comment]
@leftthomas, I do appreciate your attention in detail.  Yes, you are correct about the usage of eps to avoid division by zero. But it does affect the learning rate. At the time of writing, we didn't know that eps has such an effect on the results. Looking at how eps affect the performance, I would recommend changing the learning rate, if you insist on fixing eps to the default value. 

Regarding if the proposed methods work, I would say yes they do work for all the experiments run in the paper as eps is a constant, therefore the ablation studies are valid.

**Comment by leftthomas** [comment]
@euwern I have tested with a lot of learning rate, but they all can not reach the good results, only when fix `eps` to a big value works, so I think maybe just the `eps` takes the most effort to the results.

**Comment by euwern** [comment]
By removing all the magic numbers ((a) dividing proxies by 8, (b) gradient clipping, (c) eps of adam is set to 1.0. ), I am getting R@1=70.0±0.9 for CUB if I set the learning to 4e-6 and proxy learning rate to 4e-1.

By keeping the same learning rates, and just removing (a) and (b), I am getting R@1=72.0±1.1, which is close to the number reported in the paper (R@1=72.2±0.8), which shows that (a) and (b) has little to no effect on the performance. 

In other words, eps does have an effect on the overall performance (±2), but it is not as bad as what you have reported earlier (±20).

**Comment by leftthomas** [comment]
@euwern (a) and (b) also not mentioned in your paper, and how can you explain their effect, why you do (a) and (b)?
